packages/devtools_shared/lib/src/extensions/extension_model.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
```

Comment:
Just confirming, this is a straight move from `package:devtools_app/src/extensions/embedded/extension_model.dart` right?

Reply:
I added dart doc to the members and some validation in the factory constructor so it should be re-reviewed. I also made the json parsing for the icon code point able to handle both strings and ints